src/server.c

```diff
@@ -693,36 +693,21 @@ dictType clientDictType = {
     .no_value = 1               /* no values in this dict */
 };
 
-int htNeedsShrink(dict *dict) {
-    long long size, used;
-
-    size = dictBuckets(dict);
-    used = dictSize(dict);
-    return size > DICT_HT_INITIAL_SIZE && (used*100/size < HASHTABLE_MIN_FILL);
-}
-
-int htNeedsExpand(dict *dict) {
-    long long size, used;
-
-    size = dictBuckets(dict);
-    used = dictSize(dict);
-    return used >= size;
-}
```

Comment:
glad to see these (in server.c) go away.
note that these and the function we called after it (dictShrinkToFit) didn't seem to take the states of dict_can_resize, and dict_force_resize_ratio into account, and now we do, right?

Reply:
Yes, they are used in `dictShrinkIfNeeded` and `dictExpandIfNeeded`.